### ISSUE TITLE
[Physic - RigidBody Editor] Crash when returning to edit mode

### ISSUE THREAD

**Issue by christianclavet** [BearishSun/BansheeEngine#145]
Hi,

Been trying to see if I could make rigidbody work in the editor by setting 2 objects, one that doesnt move (floor) and one that would be in the air and fall down to it.

Setup:
1. Selected the floor model, and created a "normal" collision, done the same for the other model. 
2. Banshee show the collision icon near of the asset. Seem ok.
3. Added the rigidbody component on the floor model in the scene. Unchecked gravity, and set the mass  to 1000.
4. Added the rigidbody component on the other model and set the mass to 10. Moved the model so it's not touching the floor model.
5. Pressed "PLAY". I did not see any changed.
6. Pressed "PLAY" again to return to editor mode
7. Application crashed.
Here is a screen you can see the meshes with the collision models:
![collisions](https://user-images.githubusercontent.com/13395943/34919253-94266b14-f92e-11e7-9048-465469653918.JPG)

**EDIT:** Done some tracing of the crash with the VS2015 debugger:
Exception was risen: Access violation in write. nullptr
--> BS::RigidBody setOwner line 312
-> BS::CRigidBody::DestroyInternal() line 388
-> BS::CRigidBody::onDestroyed() line 423
...

**Comment by christianclavet** [comment]
Now since the rigidbody is in my scene, each time I try to remove the component, I face the same crash.

**Comment by BearishSun** [comment]
I managed to reproduce the crash, it was fixed with 12d82b99df66ff7d3bd820e9bff87ec54e4aa720.

In order for the physics to work you are missing Colliders. They describe a surface of an object. If you want to use a mesh as a surface (like the ones you imported), you should use a MeshCollider, and then drag and drop the mesh to the collider.

For fixed objects like walls and floors you do not need a Rigidbody. Just setting up a collider is enough.

For movable objects you need to add a Rigidbody component, as well as one or multiple Collider components (can be part of the same object as the Rigidbody, or child objects).

Note that collider meshes should be extremely simple. In fact for colliders used with rigidbodies (ones that are movable) you cannot use Normal collision meshes (those can only be used for static geometry (i.e. colliders). If you need to use a mesh for the rigidbody you must instead import it as Convex (such meshes will be limited to 255 faces and will "gift-wrap" your actual mesh). Ideally wherever possible don't use meshes at all and approximate the collision geometry using box, sphere or capsule colliders.

**Comment by christianclavet** [comment]
Hi, Just done that. No crash this time but warning coming on the console.

- I've applied a box collider component on the floor and also one on the ship.
- I've set the mass on both of them (floor is 1000 and ship is 10)
- I've applied the rigid body component on the ship set the mass to manually be 10
Pressed play to see if I would see the ship fall on the floor. Some warning in the console:
![physics](https://user-images.githubusercontent.com/13395943/34956419-f0f1db4e-f9f6-11e7-8046-83b91b3b36ee.JPG)
`"Enabling CCD on a Rigidbody but CCD is not enabled globally"`
At first nothing was moving. Looking at the console errors, one of them mentionned kinematic parameter with CCD that was not supported. I unchecked kinematic and now when I press the "PLAY" button I see the ship fall down to the floor and bounce! So the physics system is working in the editor, but the kinematic option is not working
